### ISSUE TITLE
Borders example is broken

### ISSUE THREAD

**Issue by Trashtalk217** [bevyengine/bevy#15563]
## Bevy version

main: commit 956d9ccbb1e6b240f3e434acaab9ad544df41102

## Relevant system information

```
AdapterInfo { name: "Intel(R) UHD Graphics 620 (KBL GT2)", vendor: 32902, device: 22807, device_type: IntegratedGpu, driver: "Intel open-source Mesa driver", driver_info: "Mesa 24.1.6-arch1.1", backend: Vulkan }
```

## What you did

```
cargo run --example borders
```

## What went wrong

It's supposed to look like this:
![image](https://github.com/user-attachments/assets/973446d4-976f-428f-9535-63dff52772a8)

But the rounded borders flicker out of existence immediately after loading:
![image](https://github.com/user-attachments/assets/df40cf2b-8f6f-4176-83d2-aad7806680bd)

## Additional information

I have a strong suspicion that this has to do with https://github.com/bevyengine/bevy/pull/15163, which was merged 5 days ago, so I'm pinging @ickshonpe. The fact that it's very briefly visible also makes me think that this might be a `TemporaryRenderEntity` while it is supposed to persist. I'm not sure however as I believe I've tested this before merging #15320, and it was also broken.

Actually I'm 90% sure that it's a `TemporaryRenderEntity` while it isn't supposed to be.

**Comment by kristoff3r** [comment]
I can't reproduce, it looks fine for me. I could get the bottom row to disappear if I resize my Window to be too small, but they reappear when I resize it back.

```
SystemInfo { os: "Linux 24.05 NixOS", kernel: "6.6.46", cpu: "AMD Ryzen 7 5800X 8-Core Processor", core_count: "8", memory: "47.0 GiB" }
AdapterInfo { name: "NVIDIA GeForce RTX 3070", vendor: 4318, device: 9352, device_type: DiscreteGpu, driver: "NVIDIA", driver_info: "550.78", backend: Vulkan }
```

![image](https://github.com/user-attachments/assets/bc90362a-0990-4cc6-9a93-73cf234a8d7a)

**Comment by ickshonpe** [comment]
Yes it looks fine to me too, I'm not sure why just the second set of nodes would dissappear like that.

**Comment by ickshonpe** [comment]
@Trashtalk217 does anything else disappear/reappear if you change the window size?

**Comment by alice-i-cecile** [comment]
I can't reproduce this on my Windows laptop with an integrated GPU.

> 2024-10-01T22:05:34.349941Z  INFO bevy_render::renderer: AdapterInfo { name: "Intel(R) Iris(R) Xe Graphics", vendor: 32902, device: 39497, device_type: IntegratedGpu, driver: "Intel Corporation", driver_info: "101.5768", backend: Vulkan }

![image](https://github.com/user-attachments/assets/0f9de099-3de1-4d4d-bf41-db5c53c283e6)
 
Moving and resizing the window works fine.
